Source/WebCore/animation/AnimationTimeline.h

```diff
@@ -52,6 +53,17 @@ class AnimationTimeline : public RefCounted<AnimationTimeline>, public CanMakeWe
     std::optional<double> bindingsCurrentTime();
     virtual std::optional<Seconds> currentTime() { return m_currentTime; }
 
+    virtual void detachFromDocument();
+
+    enum class ShouldUpdateAnimationsAndSendEvents : bool { No, Yes };
+    virtual ShouldUpdateAnimationsAndSendEvents documentWillUpdateAnimationsAndSendEvents() { return ShouldUpdateAnimationsAndSendEvents::Yes; }
```

Comment:
I think this should actually default to `No` and `ScrollTimeline` should override to return `Yes` if it has a valid `m_source`.

Reply:
Fixed.